src/blockdata/locktime.rs

```diff
@@ -0,0 +1,760 @@
+        mem::discriminant(self) == mem::discriminant(&other)
+    }
+
+    /// Returns true if this lock time value is a block height.
+    #[inline]
+    pub fn is_block_height(&self) -> bool {
+        match *self {
+            LockTime::Blocks(_) => true,
+            LockTime::Seconds(_) => false,
+        }
+    }
+
+    /// Returns true if this lock time value is a block time (UNIX timestamp).
+    #[inline]
+    pub fn is_block_time(&self) -> bool {
+        !self.is_block_height()
+    }
+
+    /// Returns true if this timelock constraint is satisfied by the respective `height`/`time`.
+    ///
+    /// If `self` is a blockheight based lock then it is checked against `height` and if `self` is a
+    /// blocktime based lock it is checked against `time`.
+    ///
+    /// A 'timelock constraint' refers to the `n` from `n OP_CHEKCLOCKTIMEVERIFY`, this constraint
+    /// is satisfied if a transaction with nLockTime ([`crate::Transaction::lock_time`]) set to
+    /// `height`/`time` is valid.
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use bitcoin::blockdata::locktime::{LockTime, Height, Time};
+    /// // Can be implemented if block chain data is available.
+    /// fn get_height() -> Height { todo!("return the current block height") }
+    /// fn get_time() -> Time { todo!("return the current block time") }
+    ///
+    /// let n = LockTime::from_consensus(741521); // `n OP_CHEKCLOCKTIMEVERIFY`.
+    /// if n.is_satisfied_by(get_height(), get_time()) {
+    ///     // Can create and mine a transaction that satisfies the OP_CLTV timelock constraint.
+    /// }
+    ///
+    /// // Is equivalent to:
+    ///
+    /// let is_satisfied_by = match n {
+    ///     LockTime::Blocks(height) => height >= get_height(),
+    ///     LockTime::Seconds(time) => time >= get_time(),
+    /// };
+    /// ````
+    #[inline]
+    pub fn is_satisfied_by(&self, height: Height, time: Time) -> bool {
+        use LockTime::*;
+
+        match *self {
+            Blocks(n) => n <= height,
+            Seconds(n) => n <= time,
+        }
+    }
+
+    /// Returns the inner `u32` value. This is the value used when creating this `LockTime`
+    /// i.e., `n OP_CHECKLOCKTIMEVERIFY` or nLockTime.
+    ///
+    /// # Warning
+    ///
+    /// Do not compare values return by this method. The whole point of the `LockTime` type is to
+    /// assist in doing correct comparisons. Either use `is_satisfied_by` or use the pattern below:
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use bitcoin::{LockTime, LockTime::*};
+    /// # let n = LockTime::from_consensus(100);          // n OP_CHECKLOCKTIMEVERIFY
+    /// # let lock_time = LockTime::from_consensus(100);  // nLockTime
+    ///
+    /// let is_satisfied = match (n, lock_time) {
+    ///     (Blocks(n), Blocks(lock_time)) => n <= lock_time,
+    ///     (Seconds(n), Seconds(lock_time)) => n <= lock_time,
+    ///     _ => panic!("handle invalid comparison error"),
```

Comment:
Suggesting `n.partial_cmp(&lock_time).expect("invalid comparison").is_le()` would be nicer but `is_le()` is only available in Rust 1.53, so we would have to `no_compile` this. Not sure which is better.

Reply:
I used both, added
```
    /// // Or, if you have Rust 1.53 or greater
    /// let is_satisfied = n.partial_cmp(&lock_time).expect("invalid comparison").is_le();
```

Reply:
Face palm, should have been:
```
/// // Or, if you have Rust 1.53 or greater
/// // let is_satisfied = n.partial_cmp(&lock_time).expect("invalid comparison").is_le();
```